src/app.js

```diff
@@ -1,12 +1,88 @@
+import axios from "axios";
+import distance from "euclidean-distance";
+
+// Get the API token
+async function getToken() {
+  let token;
+  try {
+    const response = await axios.post(
+      "https://blue-bottle-api-test.herokuapp.com/v1/tokens",
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    token = response.data.token;
+  } catch (e) {
+    console.error(`Exception occurred while fetching the token: ${e}`);
+  }
+  return token;
+}
+
+// Get all shop data from the API
+async function getShopData() {
+  const token = await getToken();
+
+  try {
+    const response = await axios.get(
+      `https://blue-bottle-api-test.herokuapp.com/v1/coffee_shops?token=${token}`,
+      {
+        headers: {
+          Accept: "application/json",
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    if (e.response.status == 401 || 406 || 503 || 504) {
```

Comment:
A couple of things about this line of code:

- nice that you try to handle specific error codes
- check the syntax and make sure it does what you think it does, what will it return for status 503 ? how about 504?
- some error codes are recoverable, can you think of an example?

Reply:
- I think I made a mistake about trying to check the specific status codes with the if statement. Because I noticed that if statement is always true regardless of `e.response.status`. I wanted to cover all the possible HTTP error status codes defined in the API documentation but this must have been written differently. I think getting the correct error message while testing prevented me to see the problem here.
- Because the exception message coming from Axios has the status code in it, `console.error` statement actually prints the correct status code and message. But as previously stated, it will print the correct message for other status codes which my if statement does not cover (like HTTP 404)
- I think HTTP 5xx errors (especially HTTP 504 - Gateway timeout here) can be recovered because they indicate a server problem that could potentially be temporary. HTTP 401 and 403 errors are not recoverable because they indicate that the user is not authorized to perform the action. And generally, HTTP 4xx errors indicate a client-side problem so unless the client-side is fixed, I don't think they can be automatically recovered.